app/models/campaign.rb

```diff
@@ -55,6 +55,46 @@ def last_contribution
     ).order(created_at: :desc).first&.created_at
   end
 
+  def seller_distributor_pairs
+    pairs = []
+
+    csd_pairings = CampaignsSellersDistributor
+      .joins(:campaign)
+      .where(campaigns: {
+        id: id
+      })
+
+    seller_ids = csd_pairings.map {|pairing| pairing.seller_id}
+    sellers = Seller.where(:id => seller_ids)
+    seller_id_to_seller = {}
+    sellers.each do |seller|
+      seller_id_to_seller[seller.id] = seller
+    end
+
+    distributor_ids = csd_pairings.map {|pairing| pairing.distributor_id}
+    distributors = Distributor.where(:id => distributor_ids)
+    distributor_id_to_distributor = {}
+    distributors.each do |distributor|
+      distributor_id_to_distributor[distributor.id] = distributor
+    end
+
+    csd_pairings.each do |pairing|
```

Comment:
it's technically more restful to have the caller of this endpoint make separate calls to the sellers and distributors endpoint to find this stuff

Reply:
yeah that's true but it adds complexity on the frontend and doesn't add a ton of latency for most of the campaigns. i can see the case for having clients call the api again to get sellers/distributors but in the case of campaigns, anywhere we're getting them we're showing the seller/distributor info anyways. i think having it all in 1 endpoint simplifies the frontend and doesn't unnecessarily fetch seller/distributor data for any other use case. in this case i'd prefer complexity on the backend vs frontend